packages/api/src/get-build.js

```diff
@@ -47,7 +47,23 @@ class GetBuildHandler {
 
         var metadata = await this.client.collection(global.headCollection).doc(firebaseEncode(repoid)).get();
 
-        res.send({ metadata: metadata.data(), builds: resp });
+        res.send({ metadata: metadata.data() || {}, builds: resp });
+      } catch (err) {
+        res.status(500).send({ error: err.message, stack: err.stack, name: err.name });
+      }
+    });
+
+    this.app.get('/api/repo/:orgname/:reponame/build/:buildid', async (req, res, next) => {
+      try {
+        const repoid = req.params.orgname + '/' + req.params.reponame;
+        const buildid = req.params.buildid;
+
+        const doc = await this.client.collection(global.headCollection).doc(firebaseEncode(repoid)).collection('builds').doc(firebaseEncode(buildid)).get();
+        if (doc.data()) {
+          res.send(doc.data());
+        } else {
+          throw new Error('Build not found');
```

Comment:
Any particular reason to throw an error, rather than setting the response directly? Seems to be the pattern, but I'm not familiar.

Reply:
Replaced with sending 404